scripts/inference/hf_chat.py

```diff
@@ -5,16 +5,126 @@
 import warnings
 from argparse import ArgumentParser, ArgumentTypeError, Namespace
 from contextlib import nullcontext
-from typing import Any, Dict, Tuple, Union
+from typing import Any, Dict, Union
 
 import torch
 from transformers import (AutoConfig, AutoModelForCausalLM, AutoTokenizer,
-                          PreTrainedTokenizer, PreTrainedTokenizerFast)
+                          PreTrainedTokenizer, PreTrainedTokenizerFast,
+                          StoppingCriteria, StoppingCriteriaList, TextStreamer)
 
 Tokenizer = Union[PreTrainedTokenizer, PreTrainedTokenizerFast]
 
 
-def get_dtype(dtype):
+class ChatFormatter:
+
+    def __init__(self, system: str, user: str, assistant: str) -> None:
+        self.system = system if system else '<|im_start|>system\nA conversation between a user and an LLM-based AI assistant. The assistant gives helpful and honest answers.<|im_end|>\n'
+        self.user = user if user else '<|im_start|>user\n{}<|im_end|>\n'
+        self.assistant = assistant if assistant else '<|im_start|>assistant\n{}<|im_end|>\n'
+        self.response_prefix = '<|im_start|>assistant\n'
+
+
+class Conversation:
+
+    def __init__(self, model, tokenizer: Tokenizer, chat_format: ChatFormatter,
+                 generate_kwargs: Dict[str, Any]) -> None:
+        self.model = model
+        self.tokenizer = tokenizer
+        self.chat_format = chat_format
+
+        stop_token_ids = self.tokenizer.convert_tokens_to_ids(
```

Comment:
can you add a check here that those tokens convert to a single token id? i think the rest of this code assumes that (although please correct me if im wrong)